OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -50,10 +82,33 @@ public override bool RequestTransportNotify(Actor self, Actor carryable, CPos de
 			return false;
 		}
 
+		public override bool ReserveCarryable(Actor self, Actor carryable)
+		{
+			if (State == CarryallState.Reserved)
+				UnreserveCarryable(self);
+
+			if (State != CarryallState.Idle)
+				return false;
+
+			var act = carryable.TraitOrDefault<AutoCarryable>();
+
+			if (act != null)
+			{
+				if (!act.AutoReserve(carryable, self, UnderAutoCommand))
```

Comment:
`UnderAutoCommand` variable is only used for Reserving, and it is only true when called from `RequestTransportNotify` function. So instead of tracking the carryall state `RequestTransportNotify` should send the flag directly here.

The regular `ReserveCarryable` calls actually don't need to call `AutoReserve`, `Reserve` would be just fine. Perhaps call `AutoReserve` only from `RequestTransportNotify` and don't overwrite `ReserveCarryable` at all.

Perhaps even remove the flag from `AutoReserve`, and on `AutoCarryable` just assume that when `AutoReserve` is called `UnderAutoCommand` is true.

Reply:
as I said in https://github.com/OpenRA/OpenRA/pull/20398#discussion_r1298407082, it is used for keeping
> 1. if you queue order before FerryUnit pick up the actor, then the carrier will pick up unit, deliver unit and then run the queued orders

Reply:
> UnderAutoCommand variable is only used for Reserving, and it is only true when called from RequestTransportNotify function. So instead of tracking the carryall state RequestTransportNotify should send the flag directly here.

Not only `RequestTransportNotify()`, but also`FindCarryableForTransport()` requires `UnderAutoCommand` to be true.

Reply:
I see, that's fine,  both can be handled in the same way

Reply:
add `AutoReserveCarryable` to pair the `AutoReserve` so we can drop all of the flag in `AutoCarryall`